cmd/ci-operator/main_test.go

```diff
@@ -412,3 +414,138 @@ func TestGetResolverInfo(t *testing.T) {
+				ObjectMeta: meta.ObjectMeta{
+					Name:      "console-config",
+					Namespace: "openshift-console",
+				},
+				Data: map[string]string{},
+			},
+			expectedError: fmt.Errorf("no key console-config.yaml in the data map"),
+		},
+		{
+			name: "console-config.yaml is not yaml",
+			cm: &coreapi.ConfigMap{
+				TypeMeta: meta.TypeMeta{},
+				ObjectMeta: meta.ObjectMeta{
+					Name:      "console-config",
+					Namespace: "openshift-console",
+				},
+				Data: map[string]string{
+					"console-config.yaml": "bla",
+				},
+			},
+			expectedError: fmt.Errorf("error unmarshaling JSON: json: cannot unmarshal string into Go value of type main.ConsoleConfig"),
+		},
+		{
+			name: "console-config.yaml is yaml but no cluster info at all",
+			cm: &coreapi.ConfigMap{
+				TypeMeta: meta.TypeMeta{},
+				ObjectMeta: meta.ObjectMeta{
+					Name:      "console-config",
+					Namespace: "openshift-console",
+				},
+				Data: map[string]string{
+					"console-config.yaml": "---",
+				},
+			},
+			expectedError: fmt.Errorf("no key consoleBaseAddress in the cluster info"),
+		},
+		{
+			name: "console-config.yaml is yaml and without console url in the cluster info",
+			cm: &coreapi.ConfigMap{
+				TypeMeta: meta.TypeMeta{},
+				ObjectMeta: meta.ObjectMeta{
+					Name:      "console-config",
+					Namespace: "openshift-console",
+				},
+				Data: map[string]string{
+					"console-config.yaml": `---
+clusterInfo:
+  masterPublicURL: https://api.build01.ci.devcluster.openshift.com:6443
+`,
+				},
+			},
+			expectedError: fmt.Errorf("no key consoleBaseAddress in the cluster info"),
+		},
+		{
+			name: "console-config.yaml is valid",
+			cm: &coreapi.ConfigMap{
+				TypeMeta: meta.TypeMeta{},
+				ObjectMeta: meta.ObjectMeta{
+					Name:      "console-config",
+					Namespace: "openshift-console",
+				},
+				Data: map[string]string{
+					"console-config.yaml": `apiVersion: console.openshift.io/v1
+auth:
+  clientID: console
+  clientSecretFile: /var/oauth-config/clientSecret
+  oauthEndpointCAFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+clusterInfo:
+  consoleBaseAddress: https://console-openshift-console.apps.build01.ci.devcluster.openshift.com
+  masterPublicURL: https://api.build01.ci.devcluster.openshift.com:6443
+customization:
+  branding: ocp
+  documentationBaseURL: https://docs.openshift.com/container-platform/4.3/
+kind: ConsoleConfig
+providers: {}
+servingInfo:
+  bindAddress: https://[::]:8443
+  certFile: /var/serving-cert/tls.crt
+  keyFile: /var/serving-cert/tls.key
+`,
+				},
+			},
+			expected: "https://console-openshift-console.apps.build01.ci.devcluster.openshift.com",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := getConsoleURL(tc.cm)
+			equal(t, tc.expected, actual)
+			equalError(t, tc.expectedError, err)
+		})
+	}
+}
+
+func equalError(t *testing.T, expected, actual error) {
+	if expected != nil && actual == nil || expected == nil && actual != nil {
```

Comment:
Can be simplified as `if (expected == nil ) != (actual == nil)`

Reply:
Done.